private/model/api/api.go

```diff
@@ -699,10 +699,8 @@ func New(config aws.Config) *{{ .StructName }} {
     }
 
 	{{ if .HasExternalClientConfigFields }}
-		if config.AdditionalConfig != nil {
-			if err := config.AdditionalConfig.ResolveConfig(resolveClientConfig(svc)); err != nil {
-				panic(fmt.Errorf("failed to resolve service configuration: %v", err))
-			}
+		if err := resolveClientConfig(svc, config.ConfigSources); err != nil {
+			panic(fmt.Errorf("failed to resolve service configuration: %v", err))
```

Comment:
Should we use log.Fatalf here instead of panic?

Reply:
This solution is only temporary until we implement the client builder pattern. Today the client.New(...) operation does not have the ability to return an error. So we opt for panic in this case.